pkg/config/setup/config.go

```diff
@@ -1312,10 +828,568 @@ func InitConfig(config pkgconfigmodel.Config) {
+	// Cloud Foundry
+	config.BindEnvAndSetDefault("cloud_foundry", false)
+	config.BindEnvAndSetDefault("bosh_id", "")
+	config.BindEnvAndSetDefault("cf_os_hostname_aliasing", false)
+	config.BindEnvAndSetDefault("cloud_foundry_buildpack", false)
+}
+
+func containerd(config pkgconfigmodel.Config) {
+	// Containerd
+	config.BindEnvAndSetDefault("containerd_namespace", []string{})
+	config.BindEnvAndSetDefault("containerd_namespaces", []string{}) // alias for containerd_namespace
+	config.BindEnvAndSetDefault("containerd_exclude_namespaces", []string{"moby"})
+	config.BindEnvAndSetDefault("container_env_as_tags", map[string]string{})
+	config.BindEnvAndSetDefault("container_labels_as_tags", map[string]string{})
+}
+
+func cri(config pkgconfigmodel.Config) {
+	// CRI
+	config.BindEnvAndSetDefault("cri_socket_path", "")              // empty is disabled
+	config.BindEnvAndSetDefault("cri_connection_timeout", int64(1)) // in seconds
+	config.BindEnvAndSetDefault("cri_query_timeout", int64(5))      // in seconds
+}
+
+func kubernetes(config pkgconfigmodel.Config) {
+	// Kubernetes
+	config.BindEnvAndSetDefault("kubernetes_kubelet_host", "")
+	config.BindEnvAndSetDefault("kubernetes_kubelet_nodename", "")
+	config.BindEnvAndSetDefault("eks_fargate", false)
+	config.BindEnvAndSetDefault("kubernetes_http_kubelet_port", 10255)
+	config.BindEnvAndSetDefault("kubernetes_https_kubelet_port", 10250)
+
+	config.BindEnvAndSetDefault("kubelet_tls_verify", true)
+	config.BindEnvAndSetDefault("collect_kubernetes_events", false)
+	config.BindEnvAndSetDefault("kubelet_client_ca", "")
+
+	config.BindEnvAndSetDefault("kubelet_auth_token_path", "")
+	config.BindEnvAndSetDefault("kubelet_client_crt", "")
+	config.BindEnvAndSetDefault("kubelet_client_key", "")
+
+	config.BindEnvAndSetDefault("kubernetes_pod_expiration_duration", 15*60) // in seconds, default 15 minutes
+	config.BindEnvAndSetDefault("kubelet_wait_on_missing_container", 0)
+	config.BindEnvAndSetDefault("kubelet_cache_pods_duration", 5)       // Polling frequency in seconds of the agent to the kubelet "/pods" endpoint
+	config.BindEnvAndSetDefault("kubelet_listener_polling_interval", 5) // Polling frequency in seconds of the pod watcher to detect new pods/containers (affected by kubelet_cache_pods_duration setting)
+	config.BindEnvAndSetDefault("kubernetes_collect_metadata_tags", true)
+	config.BindEnvAndSetDefault("kubernetes_metadata_tag_update_freq", 60) // Polling frequency of the Agent to the DCA in seconds (gets the local cache if the DCA is disabled)
+	config.BindEnvAndSetDefault("kubernetes_apiserver_client_timeout", 10)
+	config.BindEnvAndSetDefault("kubernetes_apiserver_informer_client_timeout", 0)
+	config.BindEnvAndSetDefault("kubernetes_map_services_on_ip", false) // temporary opt-out of the new mapping logic
+	config.BindEnvAndSetDefault("kubernetes_apiserver_use_protobuf", false)
+	config.BindEnvAndSetDefault("kubernetes_ad_tags_disabled", []string{})
```

Comment:
Removing the reference to this config in a method called `handleKubePod` prevents the extension from compiling. But since lambda is not a kube pod, then I believe we'll never hit this code anyway.